### ISSUE TITLE
Using Pre-Trained NLP Models to Interpret User Feedback

### ISSUE THREAD

**Issue by sdpetrides** [sdpetrides/ml-blog-notes#1]
### Details

[Using Pre-Trained NLP Models to Interpret User Feedback at Tubi](https://code.tubitv.com/using-pre-trained-nlp-models-to-interpret-user-feedback-at-tubi-6dffecf46510)

**Source**: Tubi

**General Topic**: An overview of how Tubi uses NLP models to categorize and learn from user feedback.

### Short Summary (200 words)

Tubi uses NLP models to learn from user feedback in two domains.

The first is through Net Promoter Scores (NPS)
 - Users give feedback with a numerical score and related comment.
 - The feedback is categorized with an NLP model into four buckets: content, app, adds, features/bugs.

The second is through customer support tickets.
 - Users submit text feedback through multiple streams.
 - The text is categorized using an NLP model into five buckets: request, issue, inquiry (non-user), complaint, question.

The mechanism for this classification uses word embeddings.
 - They tried using both Universal Sentence Encoder (USE) and Bidirectional Encoder Representations from Transformers (BERT).
 - They found that USE had better performance and better fit.

Once word embeddings are generated, they need to build classification models.
 - They started with manually labeling a set of comments. These would be the targets used in multi-class classification.
 - For classification, LightGBM was used.

They used ROC AUC as the main performance metric, while monitoring accuracy and F1 score.

The team plans to iterate on the encoder and classifier, while tracking other things like sentiment.

### Shorter Summary (100 words)

Tubi uses NLP models to learn from user feedback. The first task is to convert the text to word embeddings. Both USE and BERT were tested; USE was found to be more ideal for their use case. The next task is to use the word embeddings to classify the text into classes. They employed supervised classification using LightGBM using ROC AUC as the main performance metric. The team plans to iterate on the encoder and classifier while tracking other things like sentiment.
